helpers/mysql

```diff
@@ -118,22 +127,38 @@ ynh_mysql_drop_db() {
 
 # Dump a database
 #
-# usage: ynh_mysql_dump_db --database=database
-# | arg: -d, --database=    - the database name to dump
-# | ret: The mysqldump output
+# example: ynh_mysql_dump_db --database=roundcube --default_character_set="utf8mb4" > ./dump.sql
 #
-# example: ynh_mysql_dump_db --database=roundcube > ./dump.sql
+# usage: ynh_mysql_dump_db --database=database
+# | arg: -d, --database=                    - the database name to dump
+# | arg: -c, --default_character_set=       - the charset to use
+# | ret: the mysqldump output
 #
 # Requires YunoHost version 2.2.4 or higher.
 ynh_mysql_dump_db() {
     # Declare an array to define the options of this helper.
-    local legacy_args=d
-    local -A args_array=([d]=database=)
+    local legacy_args=dc
+    local -A args_array=( [d]=database= [c]=default_character_set= )
     local database
+    local default_character_set
     # Manage arguments with getopts
     ynh_handle_getopts_args "$@"
+    default_character_set="${default_character_set:-}"
+    MYSQL_ROOT_PWD_FILE=/etc/yunohost/mysql
 
-    mysqldump --single-transaction --skip-dump-date --routines "$database"
+    if [ -n "$default_character_set" ]
```

Comment:
Same here i think we should find a generic way in the official helper to do this. Other apps or custom instance might have the issue (cause users have seen the utf8 warning, so they might have made the change in their DB)